docs/en/sql-reference/statements/insert-into.md

```diff
@@ -56,6 +56,12 @@ INSERT INTO [db.]table [(c1, c2, c3)] SELECT ...
 
 Columns are mapped according to their position in the SELECT clause. However, their names in the SELECT expression and the table for INSERT may differ. If necessary, type casting is performed.
 
+Insert statements can have asterisk (or variants) with column transformers in the column list:
```

Comment:
This functionality is valid not only for INSERT ... SELECT ...

```
:) show create table bayonet_1

SHOW CREATE TABLE bayonet_1

┌─statement─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ CREATE TABLE default.bayonet_1
(
    `a` Int16,
    `b` String,
    `c` Int16
)
ENGINE = MergeTree()
ORDER BY a
SETTINGS index_granularity = 8192 │
└───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘

1 rows in set. Elapsed: 0.002 sec. 

clickhouse.ru-central1.internal :) insert into bayonet_1 (*) Values (1, 'a', 1)

INSERT INTO bayonet_1 (*) VALUES

Ok.

1 rows in set. Elapsed: 0.002 sec. 

:) select * from bayonet_1

SELECT *
FROM bayonet_1

┌─a─┬─b─┬─c─┐
│ 1 │ a │ 1 │
└───┴───┴───┘

1 rows in set. Elapsed: 0.002 sec. 
```

Also it has a bit tricky usage:
```

If I want to insert data in all the columns, except 'b', then I need to pass so many values how many columns I choosed in parenthesis.

```
:) insert into bayonet_1 (* EXCEPT(b)) Values (2, 2)

INSERT INTO bayonet_1 (* EXCEPT(b)) VALUES

Ok.

1 rows in set. Elapsed: 0.002 sec. 


:) select * from bayonet_1

SELECT *
FROM bayonet_1

┌─a─┬─b─┬─c─┐
│ 2 │   │ 2 │
└───┴───┴───┘
┌─a─┬─b─┬─c─┐
│ 1 │ a │ 1 │
└───┴───┴───┘

2 rows in set. Elapsed: 0.003 sec. 
```
In this example we see that the second inserted row has `a` and `c` columns filled by `values`, and `b` filled with value by default.

We definitely need examples of it the the documentation.

Reply:
@amosbird Do you have something more to expose to users about this functionality?

Reply:
Interesting. I didn't realize it either. It's also possible to construct the column list via `COLUMNS(...)` .  

>  it has a bit tricky usage. 

I don't understand what is tricky here...

Reply:
I meant that we need to accurately control the number of columns when we use wildcards in the column list constructor.

Reply:
> control the number and the type

Hmm, I think we only need to control the number here. For instance, if we do 

```
create temporary table ijk (i int, j int, k int);
insert into ijk (i, j) values (1,2,3);
insert into ijk (i, j) values (1);
```

Both would fail as expected.

With column matcher and transformer, user should have a priori that how many numbers of columns there will be, or else  insert by values doesn't make sense. So I don't think there is a confusion...